src/code/audio_load.c

```diff
@@ -1589,46 +1642,76 @@ void AudioLoad_AsyncDma(AudioAsyncLoad* asyncLoad, u32 size) {
 void AudioLoad_AsyncDmaUnkMedium(u32 devAddr, void* ramAddr, u32 size, s16 arg3) {
 }
 
-#define RELOC(v, base) (reloc = (void*)((u32)(v) + (u32)(base)))
-
-void AudioLoad_RelocateSample(SoundFontSound* sound, SoundFontData* mem, RelocInfo* relocInfo) {
+/**
+ * Read and extract information from SoundFontSound and its SoundFontSample
+ * contained in the soundFont binary loaded into ram
+ * SoundFontSound contains metadata on a sample used by a particular instrument/drum/sfx
+ * Also relocate offsets into pointers within this loaded SoundFontSound
+ *
+ * @param fontId index of font being processed
+ * @param fontData ram address of raw soundfont binary loaded into cache
+ * @param sampleBankReloc information on the sampleBank containing raw audio samples
+ */
+void AudioLoad_RelocateSample(SoundFontSound* sound, SoundFontData* fontData, SampleBankRelocInfo* sampleBankReloc) {
     SoundFontSample* sample;
     void* reloc;
 
-    if ((u32)sound->sample <= 0x80000000) {
-        sample = sound->sample = RELOC(sound->sample, mem);
-        if (sample->size != 0 && sample->unk_bit25 != 1) {
-            sample->loop = RELOC(sample->loop, mem);
-            sample->book = RELOC(sample->book, mem);
+    // Relocate an offset (relative to data loaded in ram at `base`) to a pointer (a ram address)
+#define AUDIO_RELOC(offset, base) (reloc = (void*)((u32)(offset) + (u32)(base)))
+
+    // If this has not already been relocated
+    if ((u32)sound->sample <= AUDIO_RELOCATED_ADDRESS_START) {
+
+        sample = sound->sample = AUDIO_RELOC(sound->sample, fontData);
+
+        // If the sample exists and has not already been relocated
+        // Note: this is important, as the same sample can be used by different drums, sound effects, instruments
+        if ((sample->size != 0) && (sample->isRelocated != true)) {
+            sample->loop = AUDIO_RELOC(sample->loop, fontData);
+            sample->book = AUDIO_RELOC(sample->book, fontData);
 
-            // Resolve the sample medium 2-bit bitfield into a real value based on relocInfo.
+            // Resolve the sample medium 2-bit bitfield into a real value based on sampleBankReloc.
             switch (sample->medium) {
                 case 0:
-                    sample->sampleAddr = RELOC(sample->sampleAddr, relocInfo->baseAddr1);
-                    sample->medium = relocInfo->medium1;
+                    // Relocate the offset sample within the sampleBank (not the fontData) into a pointer
+                    // (ramAddr) Overwrite the offset in fontData with this new pointer
+                    sample->sampleAddr = AUDIO_RELOC(sample->sampleAddr, sampleBankReloc->baseAddr1);
```

Comment:
So from what I read and from what Simon said, I think the result of this is a rom address, not a ram address (but I guess it could be a ram address depending on `sampleBankReloc`, just isn't in oot?)

This makes wording the documentation around a bit weird since it's not just relocating to ram anymore if this really can relocate to something other than ram...

Reply:
That's true, yes this would indeed be a rom address. So yeah, I'll reword